# Tugas Besar IF2220 Probabilitas dan Statistika



In [12]:
import pandas as pd

# Baca file Gandum.csv
columnNames = ["id", "Daerah", "SumbuUtama", "SumbuKecil", "Keunikan", "AreaBulatan", "Diameter", "KadarAir", "Keliling", "Bulatan", "Ransum", "Kelas"]
df = pd.read_csv("Gandum.csv", names=columnNames)
df.describe()

id       Daerah  SumbuUtama  SumbuKecil    Keunikan  \
count  500.000000   500.000000  500.000000  500.000000  500.000000   
mean   250.500000  4801.246000  116.045171   53.715246    0.878764   
std    144.481833   986.395491   18.282626    4.071075    0.036586   
min      1.000000  2522.000000   74.133114   39.906517    0.719916   
25%    125.750000  4042.750000  104.116098   51.193576    0.863676   
50%    250.500000  4735.000000  115.405140   53.731199    0.890045   
75%    375.250000  5495.500000  129.046792   56.325158    0.907578   
max    500.000000  7453.000000  227.928583   68.977700    0.914001   

       AreaBulatan    Diameter    KadarAir    Keliling     Bulatan  \
count   500.000000  500.000000  500.000000  500.000000  500.000000   
mean   4937.048000   77.771158    0.648372  281.479722    0.761737   
std    1011.696255    8.056867    0.094367   37.335402    0.061702   
min    2579.000000   56.666658    0.409927  197.015000    0.174590   
25%    4170.250000   71.745308    0.572632  255.883000    0.731991   
50%    4857.000000   77.645277    0.626117  280.045500    0.761288   
75%    5654.250000   83.648598    0.726633  306.062500    0.796361   
max    7720.000000   97.413830    0.878899  488.837000    0.904748   

           Ransum       Kelas  
count  500.000000  500.000000  
mean     2.150915    1.502000  
std      0.249767    0.500497  
min      1.440796    1.000000  
25%      1.983939    1.000000  
50%      2.193599    2.000000  
75%      2.381612    2.000000  
max      2.464809    2.000000

1. Menulis deskripsi statistika (Descriptive Statistics) dari semua kolom pada data yang bersifat
numerik, terdiri dari mean, median, modus, standar deviasi, variansi, range, nilai minimum,
maksimum, kuartil, IQR, skewness dan kurtosis. Boleh juga ditambahkan deskripsi lain.

In [56]:
def getStatisticDescription(df):
    print("Mean =", df.mean())
    print("Median =", df.median())
    modus = df.mode()
    print("Modus = ", end="")
    for i in range(len(modus)):
        if i != 0:
            print(", ", end="")
        print(modus[i], end="")
    print()
    print("Standar Deviasi =", df.std())
    print("Variansi =", df.var())
    print("Range =", df.max()-df.min())
    print("Nilai Minimum =", df.min())
    print("Nilai Maksimum =", df.max())
    print("Q1 =", df.quantile(0.25))
    print("Q2 =", df.quantile(0.5))
    print("Q3 =", df.quantile(0.75))
    print("IQR =", df.quantile(0.75)-df.quantile(0.25))
    print("Skewness =", df.skew())
    print("Kurtosis =", df.kurt())

### Kolom Daerah

In [58]:
getStatisticDescription(df["Daerah"])

Mean = 4801.246
Median = 4735.0
Modus = 3992, 4881, 5642, 6083
Standar Deviasi = 986.3954914816017
Variansi = 972976.0656152305
Range = 4931
Nilai Minimum = 2522
Nilai Maksimum = 7453
Q1 = 4042.75
Q2 = 4735.0
Q3 = 5495.5
IQR = 1452.75
Skewness = 0.23814408738280812
Kurtosis = -0.4346305340273977


2. Membuat Visualisasi plot distribusi, dalam bentuk histogram dan boxplot untuk setiap
kolom numerik. Berikan uraian penjelasan kondisi setiap kolom berdasarkan kedua plot
tersebut.

3. Menentukan setiap kolom numerik berdistribusi normal atau tidak. Gunakan normality test
yang dikaitkan dengan histogram plot

4. Melakukan test hipotesis 1 sampel, dengan menuliskan 6 langkah testing dan menampilkan
juga boxplotnya untuk kolom/bagian yang bersesuaian.

    a. Nilai rata-rata Daerah di atas 4700?

    b. Nilai Rata-rata Sumbu Utama tidak sama dengan 116?

    c. Nilai Rata-rata 20 baris pertama kolom Sumbu Kecil bukan 50?

    d. Proporsi nilai Diameter yang lebih dari 85, adalah tidak sama dengan 15% ?

    e. Proporsi nilai Keliling yang kurang dari 100, adalah kurang dari 5% ?

5. Melakukan test hipotesis 2 sampel, dengan menuliskan 6 langkah testing dan menampilkan
juga boxplotnya untuk kolom/bagian yang bersesuaian.

    a. Data kolom AreaBulatan dibagi 2 sama rata: bagian awal dan bagian akhir kolom. Benarkah rata-rata kedua bagian tersebut sama?

    b. Data kolom Kadar Air dibagi 2 sama rata: bagian awal dan bagian akhir kolom. Benarkah rata-rata bagian awal lebih besar dari pada bagian akhir sebesar 0.2?

    c. Rata-rata 20 baris pertama kolom Bulatan sama dengan 20 baris terakhirnya?

    d. Proporsi nilai bagian awal Ransum yang lebih dari 2, adalah lebih besar daripada, proporsi nilai yang sama di bagian akhir Ransum?

    e. Bagian awal kolom Diameter memiliki variansi yang sama dengan bagian akhirnya?

6. Test korelasi: tentukan apakah setiap kolom non-target berkorelasi dengan kolom target,
dengan menggambarkan juga scatter plot nya. Gunakan correlation test.